sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala

```diff
@@ -271,6 +271,10 @@ class ParquetToSparkSchemaConverter(
             } else {
               TimestampNTZType
             }
+          // SPARK-40819: NANOS are not supported as a Timestamp, convert to LongType without
+          // timezone awareness to address behaviour regression introduced by SPARK-34661
```

Comment:
Can we do a truncation and still read it as timestamp type?

Reply:
I do not think this is a good idea as the precision will be lost which is extremely important for high frequency time series. 

I haven’t verified but end users/developers would still be able to `.cast(Timestamp)` which I believe would truncate the timestamp; allowing developers to make that decision makes more sense then forcing the loss of precision.

Reply:
Yes, the main purpose of the nano timestamp is the nano precision.